### DISCUSSION TITLE
[Deployment] Issue deploying Account Abstraction Factory via EthersJS.

### DISCUSSION THREAD

**Discussion by cryptobank04** [zkSync-Community-Hub/zksync-developers#251] [Deployments] [answered]
### Environment

Mainnet

### zkSolc Version

n/a

### zksync-web3 Version

^5.7.2

### Hardhat.config.ts

```javascript
import { HardhatUserConfig } from 'hardhat/config';

import '@matterlabs/hardhat-zksync-node';
import '@matterlabs/hardhat-zksync-deploy';
import '@matterlabs/hardhat-zksync-solc';
import '@matterlabs/hardhat-zksync-verify';

const config: HardhatUserConfig = {
  defaultNetwork: 'zkSyncSepoliaTestnet',
  networks: {
    zkSyncSepoliaTestnet: {
      url: 'https://sepolia.era.zksync.dev',
      ethNetwork: 'sepolia',
      zksync: true,
      verifyURL:
        'https://explorer.sepolia.era.zksync.dev/contract_verification',
    },
    zkSyncMainnet: {
      url: 'https://mainnet.era.zksync.io',
      ethNetwork: 'mainnet',
      zksync: true,
      verifyURL:
        'https://zksync2-mainnet-explorer.zksync.io/contract_verification',
    },
    zkSyncGoerliTestnet: {
      // deprecated network
      url: 'https://testnet.era.zksync.dev',
      ethNetwork: 'goerli',
      zksync: true,
      verifyURL:
        'https://zksync2-testnet-explorer.zksync.dev/contract_verification',
    },
    dockerizedNode: {
      url: 'http://localhost:3050',
      ethNetwork: 'http://localhost:8545',
      zksync: true,
    },
    inMemoryNode: {
      url: 'http://127.0.0.1:8011',
      ethNetwork: '', // in-memory node doesn't support eth node; removing this line will cause an error
      zksync: true,
    },
    hardhat: {
      zksync: true,
    },
  },
  zksolc: {
    version: 'latest',
    settings: {
      // find all available options in the official documentation
      // https://era.zksync.io/docs/tools/hardhat/hardhat-zksync-solc.html#configuration
    },
  },
  solidity: {
    version: '0.8.17',
  },
};

export default config;
```

### Deployment Script (WITHOUT PRIVATE KEY)

```javascript
deployAAFactory = async () => {
    const aaFactory = new zksync.ContractFactory(
      aafactory_artifact.abi,
      aafactory_artifact.bytecode,
      this.account.signer
    );

    const aaFacotryContract = await aaFactory.deploy(
      zksync.utils.hashBytecode(aaFactory.bytecode)
    );

    await aaFacotryContract.deployTransaction.wait();

    console.log('✅ AA Contract deployed to: ', aaFacotryContract.address);

    return aaFacotryContract;
  };
```

### Package.json

```json
n/a
```

### Contract Code

```solidity
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.13;

import "@matterlabs/zksync-contracts/l2/system-contracts/Constants.sol";
import "@matterlabs/zksync-contracts/l2/system-contracts/libraries/SystemContractsCaller.sol";

contract AAFactory {
    bytes32 public aaBytecodeHash;

    constructor(bytes32 _aaBytecodeHash) {
        aaBytecodeHash = _aaBytecodeHash;
    }

    function deployAccount(
        bytes32 salt,
        address owner
    ) external returns (address accountAddress) {
        (bool success, bytes memory returnData) = SystemContractsCaller
            .systemCallWithReturndata(
                uint32(gasleft()),
                address(DEPLOYER_SYSTEM_CONTRACT),
                uint128(0),
                abi.encodeCall(
                    DEPLOYER_SYSTEM_CONTRACT.create2Account,
                    (
                        salt,
                        aaBytecodeHash,
                        abi.encode(owner),
                        IContractDeployer.AccountAbstractionVersion.Version1
                    )
                )
            );
        require(success, "Deployment failed");

        (accountAddress) = abi.decode(returnData, (address));
    }
}
```

### Does this work on other EVMs? (If yes, please list at least 1 of them)

not sure.

### Description of What Your Contract Does

This is the AAFactory from the Daily Spend Limit tutorial.
https://era.zksync.io/docs/dev/tutorials/aa-daily-spend-limit.html

### Additional Details

I am following the Daily Spend Limit tutorial from this link:
https://era.zksync.io/docs/dev/tutorials/aa-daily-spend-limit.html

I am modifying the tutorial by deploying using ethersjs instead. I am using the `zksync.ContractFactory` to try and deploy the AA Factory, however I keep getting this error.

```
Error: unsupported transaction type: 113 (operation="serializeTransaction", transactionType=113, code=UNSUPPORTED_OPERATION, version=transactions/5.7.0)
```

Any idea how i can resolve this?

**Comment by Romsters** [comment, marked as answer]
Hi @cryptobank04 , was not able to reproduce this exact error, but there a couple of things I see you need to change:
- Add `isSystem` flag to your `zksolc` config. Please check out this [doc](https://era.zksync.io/docs/reference/architecture/system-contracts.html#protected-access-to-some-of-the-system-contracts) for more info.
- use Account Abstraction contract bytecode hash instead of the factory bytecode hash when calling `deploy` on the factory.
```js
const aaFacotryContract = await aaFactory.deploy(
      zksync.utils.hashBytecode(aaContract.bytecode)
    );
```

**Comment by belanoparo** [comment]
The error "unsupported transaction type: 113" indicates that the zkSync transaction type being used is not supported by the current version of zkSync-js. This could be due to a mismatch between the version of zkSync-js being used and the transaction type being generated.

To resolve this issue, you can try the following steps:

Update the version of zkSync-js to the latest version that supports the transaction type you're trying to use.
Check the zkSync-js documentation or release notes to verify which transaction types are supported in the version you're using.
Ensure that the bytecode and ABI of your contract are compatible with the version of zkSync-js you're using.
If the issue persists, consider reaching out to the zkSync community or opening an issue on their GitHub repository for further assistance.
By ensuring compatibility between your contract, zkSync-js version, and transaction type, you should be able to resolve the "unsupported transaction type" error.

**Comment by sarportzapbisusan** [comment]
The error "unsupported transaction type: 113" suggests that there is an issue with the transaction type being used. This error may occur due to incompatible transaction types or versions between the client library (zksync-web3) and the zksync server. To resolve this, ensure that the client library and server are compatible and up-to-date. Additionally, verify that the contract deployment code and parameters are correct.

**Comment by lykinsjennifer** [comment]
The error "unsupported transaction type: 113" typically indicates that the transaction type being used is not supported by the network or the library being used. Ensure that the transaction type and parameters are correct for the zkSync network you are interacting with. Additionally, consider checking for any updates or documentation changes in the zkSync library that may address this issue.

**Comment by knowenlophoroger** [comment]
The user is using zksync-web3 version ^5.7.2 and has provided their hardhat.config.ts file. They have not specified any issues or errors they are encountering, and have not provided any additional details or repository links.
